mono/mini/method-to-ir.c

```diff
@@ -9099,21 +9100,8 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 					ins->inst_p0 = cmethod;
 					ins->inst_p1 = arg_array [0];
 					MONO_ADD_INS (cfg->cbb, ins);
-					link_bblock (cfg, cfg->cbb, end_bblock);
-					start_new_bblock = 1;
-
-					// FIXME: Eliminate unreachable epilogs
 
-					/*
-					 * OP_TAILCALL has no return value, so skip the CEE_RET if it is
-					 * only reachable from this call.
-					 */
-					GET_BBLOCK (cfg, tblock, ip + 5);
-					if (tblock == cfg->cbb || tblock->in_count == 0)
-						skip_ret = TRUE;
-					push_res = FALSE;
-
-					goto call_end;
+					goto tailcall_remove_ret;
 				}
```

Comment:
The way this is currently done is we have one call_end label, and a bunch of flags are set to control its behavior, you can just add 1 more flag.

Reply:
An option or insist? I guess I call it tailcall_remove_ret, move the label up, use if on that gboolean.

Reply:
@vargaz Commit 94b5bb4 is an attempt to do what you say.